### PR TITLE
Astyle regression check (this X is because of a Makefile change)-->

### PR DESCRIPTION
**Pull request by kevingranade**
Documentation update, grabbed some low-hanging astyle fruit, and added an astyle regression check to the release build.
Once this is in, PRs that cause regressions in already-astyled files will fail their builds immediately (assuming Jenkins has a recent enough version of astyle installed, we'll see).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,16 +18,7 @@ There are a couple of guidelines we suggest sticking to:
 
 ## Code Style
 
-Current policy is to only update code to the standard style when changing a substantial portion of it, but **please** do this in a separate commit. Blocks of code can be passed through astyle
```
Comment (kevingranade) [inline CONTRIBUTING.md]: Having this duplicated here didn't really add any value, if you want to read about styling you can just read CODE_STYLE.txt

File: doc/CODE_STYLE.txt [hunk 2]
```diff
@@ -37,3 +37,21 @@ int foo( int arg1, int *arg2 )
...
+Here's a helpful workflow to astyle some low-hanging fruit:
+# Astyle all the things.
+make astyle-all
+# List the changed files.
+# |             Sort by number of lines changed.
+# |             |           Truncate away the first part of the list.
+# |             |           |           Truncate away the summary line.
+# |             |           |           |           Trim off the fields except the filename.
+# |             |           |           |           |                 Revert the listed files to the upstream version.
+git diff --stat | sort -k 3 | tail -201 | head -200 | cut -d ' ' -f 2 | xargs git checkout
```
Comment (kevingranade) [inline doc/CODE_STYLE.txt]: Learn BASH, you too can write insane monstrosities like this!

File: src/lua_console.cpp [hunk 3]
```diff
@@ -24,9 +24,24 @@ std::string lua_console::get_input()
     long key = 0;
     int pos = -1;
     std::map<long, std::function<void()>> callbacks {
-        { KEY_ESCAPE, [this]() { this->quit(); } },
-        { KEY_NPAGE, [this]() { this->scroll_up(); } },
-        { KEY_PPAGE, [this]() { this->scroll_down(); } } };
+        {
```
Comment (kevingranade) [inline src/lua_console.cpp]: I'm not super happy with how large this ends up being, hopefully a later astyle release will have better handling for initializers.

**Comment by sparr** [comment]
I am super amused that Jenkins refuses to test this :)

**Pull request merged**